ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/message/InsertHandler.java

```diff
@@ -17,17 +17,41 @@
  */
 package org.apache.hadoop.hive.ql.parse.repl.load.message;
 
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hive.metastore.messaging.InsertMessage;
 import org.apache.hadoop.hive.ql.exec.Task;
+import org.apache.hadoop.hive.ql.parse.EximUtil;
+import org.apache.hadoop.hive.ql.parse.ReplicationSpec;
 import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.apache.hadoop.hive.ql.parse.repl.load.MetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.Serializable;
+import java.util.Collections;
 import java.util.List;
 
 public class InsertHandler extends AbstractMessageHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(InsertHandler.class);
+
   @Override
   public List<Task<? extends Serializable>> handle(Context withinContext)
       throws SemanticException {
+    try {
+      FileSystem fs =
+          FileSystem.get(new Path(withinContext.location).toUri(), withinContext.hiveConf);
+      MetaData metaData =
+          EximUtil.readMetaData(fs, new Path(withinContext.location, EximUtil.METADATA_NAME));
+      ReplicationSpec replicationSpec = metaData.getReplicationSpec();
+      if (replicationSpec.isNoop()) {
+        return Collections.emptyList();
+      }
+    } catch (Exception e) {
+      LOG.error("failed to load insert event", e);
+      throw new SemanticException(e);
+    }
+
```

Comment:
This looks like some change unrelated to external table replication. Something like fixing an existing bug.

Reply:
nope this is for making sure that insert events for external tables are no-op

Reply:
Ok. Thanks for the clarification.